src/libs/ajax/leonardo/models/runtime-config-models.ts

```diff
@@ -73,3 +75,7 @@ export const isGceConfig = (config: RuntimeConfig): config is GceConfig => {
 };
 
 export const isAzureConfig = (config: RuntimeConfig): config is AzureConfig => config.cloudService === 'AZURE_VM';
+
+export type NormalizedRuntimeConfig = {
```

Comment:
Would it be possible to slip `normalizedRegion` into the base `RuntimeConfig`, so that `RuntimeConfig` is just always normalized? Or are there other factors preventing that?

Reply:
I don't want to mutate the original return from the API, only add to it

Reply:
Adding a field is a mutation but not a lossy one. Is that much different?

Reply:
Yes, a mutation that is not a lossy one is a decoration, and thus the model is backwards compatible. In other words, a runtimeConfig that is decorated can be used in the code as a runtime config that is not decorated, but the reverse isn't true.

Reply:
So isn't it pretty safe to decorate RuntimeConfig directly? Like, rename existing RuntimeConfig to RawRuntimeConfig, and then make a new RuntimeConfig which is the decorated version, for use everywhere except the API layer?

Reply:
We came up with a solution that provided stronger type safety and backwards compatibility, see the updated version of this file